include/libtorrent/alert_types.hpp

```diff
@@ -2569,11 +2569,50 @@ namespace libtorrent {
 		virtual std::string message() const override;
 	};
 
+	struct TORRENT_EXPORT dht_sample_infohashes_alert final : alert
+	{
+		dht_sample_infohashes_alert(aux::stack_allocator& alloc
+			, udp::endpoint const& endp
+			, time_duration interval
+			, int num
+			, std::vector<sha1_hash> const& samples
+			, std::vector<std::pair<sha1_hash, udp::endpoint>> const& nodes);
+
+		static const int static_category = alert::dht_operation_notification;
+		TORRENT_DEFINE_ALERT(dht_sample_infohashes_alert, 93)
+
+		virtual std::string message() const override;
+
+		udp::endpoint const endpoint;
+
+		time_duration const interval;
+
+		// This field indicates how many infohash keys are currently in the node's storage.
+		// If the value is larger than the number of returned samples it indicates that the
+		// indexer may obtain additional samples after waiting out the interval.
+		int const num_infohashes;
+
+		int num_samples() const;
+		std::vector<sha1_hash> samples() const;
+
+		int num_nodes() const;
+		std::vector<std::pair<sha1_hash, udp::endpoint>> nodes() const;
```

Comment:
would you mind documenting the ``nodes()`` function too? As far as I understand, this returns the *node-id* and endpoints for more DHT nodes returned by the request.